code/modules/mapping/mapping_helpers.dm

```diff
@@ -380,3 +381,37 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 	log_mapping("[src] at [x],[y] could not find an airlock on current turf, cannot place paper note.")
 	qdel(src)
 
+//This helper applies traits to things on the map directly.
+/obj/effect/mapping_helpers/trait_injector
+	name = "Trait Injector"
+	icon_state = "trait"
+	late = TRUE
+	///Will inject into all fitting the criteria if true, otherwise first found.
+	var/all = FALSE
+	///Will inject into atoms of this type.
+	var/target_type
+	///Will inject into atoms with this name.
+	var/target_name
+	///Name of the trait, in the lower-case text (NOT the upper-case define) form.
+	var/trait_name
+
+//Late init so everything is likely ready and loaded (no warranty)
+/obj/effect/mapping_helpers/trait_injector/LateInitialize()
+	if(!GLOB.trait_name_map)
+		GLOB.trait_name_map = generate_trait_name_map()
+	if(!GLOB.trait_name_map.Find(trait_name))
+		CRASH("Wrong trait in [type] - [trait_name] is not a trait")
+	var/turf/target_turf = get_turf(src)
+	for(var/atom/atom_on_turf in target_turf.GetAllContents())
+		if(atom_on_turf == src)
```

Comment:
```suggestion
	for(var/a in target_turf.GetAllContents())
		var/atom/atom_on_turf = a
		if(atom_on_turf == src)
```
No need to type check

Reply:
it looks cooler though when we do it in one line

Reply:
It also hides `null` entries in lists and performs unnecessary `istype()` checks.

Reply:
ok fine...